tests/Integrations/AMQP/AMQPTest.php

```diff
@@ -0,0 +1,837 @@
+                Tag::RABBITMQ_EXCHANGE          => 'basic_get_test',
+                Tag::RABBITMQ_ROUTING_KEY       => '<all>'
+            ]),
+            SpanAssertion::build(
+                'amqp.basic.publish',
+                'amqp',
+                'queue',
+                'basic.publish basic_get_test -> <all>'
+            )->withExactTags([
+                Tag::SPAN_KIND                  => 'producer',
+                Tag::COMPONENT                  => 'amqp',
+                Tag::MQ_SYSTEM                  => 'rabbitmq',
+                Tag::MQ_DESTINATION_KIND        => 'queue',
+                Tag::MQ_PROTOCOL                => 'AMQP',
+                Tag::MQ_PROTOCOL_VERSION        => AMQPChannel::getProtocolVersion(),
+                Tag::MQ_MESSAGE_PAYLOAD_SIZE    => 12,
+                Tag::MQ_OPERATION               => 'send',
+                Tag::RABBITMQ_ROUTING_KEY       => '<all>',
+                Tag::RABBITMQ_EXCHANGE          => 'basic_get_test',
+                Tag::RABBITMQ_DELIVERY_MODE     => '2'
+            ]),
+            SpanAssertion::build(
+                'amqp.basic.get',
+                'amqp',
+                'queue',
+                'basic.get basic_get_queue'
+            )->withExactTags([
+                Tag::SPAN_KIND                  => 'consumer',
+                Tag::COMPONENT                  => 'amqp',
+                Tag::MQ_SYSTEM                  => 'rabbitmq',
+                Tag::MQ_DESTINATION_KIND        => 'queue',
+                Tag::MQ_PROTOCOL                => 'AMQP',
+                Tag::MQ_PROTOCOL_VERSION        => AMQPChannel::getProtocolVersion(),
+                Tag::MQ_DESTINATION             => 'basic_get_queue',
+                Tag::MQ_MESSAGE_PAYLOAD_SIZE    => 12,
+                Tag::MQ_OPERATION               => 'receive',
+                Tag::RABBITMQ_ROUTING_KEY       => '<all>',
+                Tag::RABBITMQ_EXCHANGE          => 'basic_get_test',
+                Tag::RABBITMQ_DELIVERY_MODE     => '2'
+            ]),
+            SpanAssertion::build(
+                'amqp.basic.ack',
+                'amqp',
+                'queue',
+                'basic.ack 1'
+            )->withExactTags([
+                Tag::SPAN_KIND                  => 'process',
+                Tag::COMPONENT                  => 'amqp',
+                Tag::MQ_SYSTEM                  => 'rabbitmq',
+                Tag::MQ_DESTINATION_KIND        => 'queue',
+                Tag::MQ_PROTOCOL                => 'AMQP',
+                Tag::MQ_PROTOCOL_VERSION        => AMQPChannel::getProtocolVersion()
+            ]),
+        ]);
+    }
+
+    public function testDistributedTracing()
+    {
+        // Note: This test is extremely flaky, locally at least. It will eventually pass with some tries...
+        // Reason: We may parse the traces from dumped data BEFORE the traces are flushed.
```

Comment:
Can you elaborate? Also `\DDTrace\Tests\Common\TracerTestTrait::resetRequestDumper()` is a thing, if needed.

Reply:
For me it was never flaky, so, let's ship it.